src/signature.js

```diff
@@ -0,0 +1,128 @@
+import crypto from "crypto";
+import fetch from "node-fetch";
+
+import { account, domain } from "./util.js";
+import { getPrivateKey } from "./activity-pub-db.js";
+
+/**
+ * Returns base-64 encoded SHA-256 digest of provided data
+ *
+ * @param {string} data - UTF-8 string to be hashed
+ *
+ * @returns {string}
+ */
```

Comment:
thanks for all the JSDoc annotations! I should probably spend a few nights doing the same; it would make dev easier now that other people are looking at the code (and would probably expose a few weird decisions that are in there already)